modules/core/src/main/java/org/apache/synapse/mediators/eip/splitter/CloneMediator.java

```diff
@@ -135,6 +149,32 @@ public boolean mediate(MessageContext synCtx) {
         return continueParent;
     }
 
+    private void executeTargetIterations(MessageContext synCtx) {
+        int noOfIterations = resolveIterationsCount(synCtx);
+        SynapseLog synLog = getLog(synCtx);
+        Target target = targets.get(0);
+        for (int i = 0; i < noOfIterations; ++i) {
+            if (synLog.isTraceOrDebugEnabled()) {
+                synLog.traceOrDebug("Submitting " + i + " of " + noOfIterations +
+                        " messages for " + (isSequential() ? "sequential processing" : "parallel processing"));
+            }
+            synCtx.setProperty(ITERATION_INDEX_PROPERTY_NAME, i);
+            MessageContext clonedMsgCtx = getClonedMessageContext(synCtx, i, noOfIterations);
+            ContinuationStackManager.addReliantContinuationState(clonedMsgCtx, i, getMediatorPosition());
+            target.mediate(clonedMsgCtx);
+        }
+    }
+
+    private int resolveIterationsCount(MessageContext synCtx) {
+        String countStr = "";
+        if (getDynamicIterationsValue() != null) {
+            countStr = getDynamicIterationsValue().evaluateValue(synCtx);
+        } else if (getIterations() != null) {
```

Comment:
do we need an else if, isn't only else is enough. i.e if getDynamicIterationsValue() == null, we should get from the other.

Reply:
+1